app/static/js/index.js

```diff
@@ -9,3 +9,16 @@ $(document).on('click', '.preference .delete', function () {
         }
     });
 });
+
+$(document).ready(function () {
+    $("#add-preference-form").submit(function (event) {
+        $.post($(this).attr('action'), $(this).serialize(), function (res) {
+            $prefList = $("#preferences > ul")
+            $newItem = $("<li>").addClass('preference').data('id', res.id)
```

Comment:
Not something you need to change, but this is one of the things HTML5 template tags are meant for.